cpp/cmake_modules/GandivaBuildUtils.cmake

```diff
@@ -91,7 +90,7 @@ function(add_gandiva_unit_test REL_TEST_NAME)
     ${CMAKE_SOURCE_DIR}/src
   )
   target_link_libraries(${TEST_NAME}
-    PRIVATE arrow_shared ${GANDIVA_TEST_LINK_LIBS} Boost::boost
+    PRIVATE arrow_shared ${GANDIVA_TEST_LINK_LIBS}
   )
   add_test(NAME ${TEST_NAME} COMMAND ${TEST_NAME})
   set_property(TEST ${TEST_NAME} PROPERTY LABELS gandiva,unittest ${TEST_NAME})
```

Comment:
We should endeavor to remove this file (or merge with `gandiva/CMakeLists.txt`) before the next release

Reply:
OK.